libraries/c_sdk/standard/mqtt/src/iot_mqtt_operation.c

```diff
@@ -498,8 +499,17 @@ bool _IotMqtt_DecrementOperationReferences( _mqttOperation_t * pOperation,
         EMPTY_ELSE_MARKER;
     }
 
-    /* Decrement job reference count. */
-    if( taskPoolStatus == IOT_TASKPOOL_SUCCESS )
+    /*
+     * Shim Implementation is using taskpool only for CONNECT operation so this logic is updated.
```

Comment:
This statement can be removed. I think you are providing enough explanations in the 2 cases below.